lib/instrumentation/undici.js

```diff
@@ -133,17 +144,25 @@ module.exports = function addUndiciChannels(agent, undici, modName, shim) {
     const url = new URL(urlString)
 
     const name = NAMES.EXTERNAL.PREFIX + url.host + url.pathname
-    const segment = shim.createSegment(name, recordExternal(url.host, 'undici'), parentSegment)
+    let segment
+    if (request[SYMBOLS.SEGMENT]) {
```

Comment:
i'd follow the logic we have in other hook points where we do this and add it to the top of this hook `request:create`
```js
    const activeSegment = request[SYMBOLS.SEGMENT]
    if (!activeSegment) {
      return
    }
```